File: src/ca/uqam/mgl7460/tp1/tests/CreationDemandesPret.java
package ca.uqam.mgl7460.tp1.tests;

import ca.uqam.mgl7460.tp1.types.modeles.Adresse;
import ca.uqam.mgl7460.tp1.types.modeles.DemandePret;
import ca.uqam.mgl7460.tp1.types.modeles.DemandeurPret;
import ca.uqam.mgl7460.tp1.types.modeles.Propriete;
import ca.uqam.mgl7460.tp1.types.modeles.ProvinceOuTerritoire;
import ca.uqam.mgl7460.tp1.types.utils.Fabrique;

public class CreationDemandesPret {

    private static Fabrique fabrique = Fabrique.getSingletonFabrique();

    private CreationDemandesPret() {}

    public static DemandePret getDemandePretOuiOuiOui() {
        DemandeurPret demandeur = fabrique.creerDemandeurPret("Jeanne","Bergeron","123-456-789",120000f,40000f,765);
        Propriete propriete = fabrique.creerPropriete(new Adresse("201A", "2100", "Saint-Urbain", "Montréal", ProvinceOuTerritoire.QUEBEC, "H2X2XH"),400000f);
        return fabrique.creerDemandePret(propriete, demandeur,405000f,30000f);
    }

    public static DemandePret getDemandePretOuiOuiNon() {
        DemandeurPret demandeur = fabrique.creerDemandeurPret("Mamadou","Diallo","321-654-987",120000f,40000f,765);
        Propriete propriete = fabrique.creerPropriete(new Adresse("1504", "1800", "Bleury", "Montréal", ProvinceOuTerritoire.QUEBEC, "H2Y2Y5"),400000f);
        return fabrique.creerDemandePret(propriete, demandeur,405000f,5000f);
    }

    public static DemandePret getDemandePretOuiNonNon() {
        DemandeurPret demandeur = fabrique.creerDemandeurPret("Jorge","Riviera","987-654-123",150000f,50000f,780);
        Propriete propriete = fabrique.creerPropriete(new Adresse("1504", "1800", "Bleury", "Calgary", ProvinceOuTerritoire.ALBERTA, "J3H3G5"),400000f);
        return fabrique.creerDemandePret(propriete, demandeur,405000f,5000f);
    }

    public static DemandePret getDemandePretNonNonNon() {
        DemandeurPret trump = fabrique.creerDemandeurPret("Donald","Trump", "666-666-666",600000f,83000000f,520);
        Propriete trumpTower = fabrique.creerPropriete(new Adresse("", "325", "Bay Street", "Toronto", ProvinceOuTerritoire.ONTARIO, "M5H 4G3"),100000f);
        return fabrique.creerDemandePret(trumpTower, trump,45000000f,5000f);
    }

    public static DemandePret getDemandePretOuiNonOui() {
        DemandeurPret legault = fabrique.creerDemandeurPret("François","Legault", "555-555-555",275000,75000f,765);
        Propriete siegeNorthVolt = fabrique.creerPropriete(new Adresse("", "20", "Alströmergatan", "Stockholm", ProvinceOuTerritoire.NEWFOUNDLAND_AND_LABRADOR, "M5H 4G3"),480000000f);
        return fabrique.creerDemandePret(siegeNorthVolt, legault,450000000f,275000000f);
    }

}

File: src_python/tests/creation_demandes_pret.py
from utils import fabrique
from modeles.adresse import Adresse
from modeles.province_ou_territoire import ProvinceOuTerritoire

def demande_pret_oui_oui_oui():
     demandeur = fabrique.creer_demandeur_pret(prenom="Jeanne", nom="Bergeron", nas="123-456-789", revenu_annuel=120000.0, obligations_annuelles=40000.0, score_credit=765)
     propriete = fabrique.creer_propriete(adresse=Adresse(numero_porte="201A", numero_rue="2100", nom_rue="Saint-Urbain", ville="Montréal", province=ProvinceOuTerritoire.QUEBEC, code_postal="H2X2XH"), valeur_marche=400000.0)
     return fabrique.creer_demande_pret(propriete=propriete, demandeur_pret=demandeur,prix_achat=405000.0, montant_mise_de_fonds=30000.0)

def demande_pret_oui_oui_non():
    demandeur = fabrique.creer_demandeur_pret(
        prenom="Mamadou",
        nom="Diallo",
        nas="321-654-987",
        revenu_annuel=120000.0,
        obligations_annuelles=40000.0,
        score_credit=765
    )

    propriete = fabrique.creer_propriete(
        adresse=Adresse(
            numero_porte="1504",
            numero_rue="1800",
            nom_rue="Bleury",
            ville="Montréal",
            province=ProvinceOuTerritoire.QUEBEC,
            code_postal="H2Y2Y5"
        ),
        valeur_marche=400000.0
    )

    return fabrique.creer_demande_pret(
        propriete=propriete,
        demandeur_pret=demandeur,
        prix_achat=405000.0,
        montant_mise_de_fonds=5000.0
    )

def demande_pret_oui_non_non():
    demandeur = fabrique.creer_demandeur_pret(
        prenom="Jorge",
        nom="Riviera",
        nas="987-654-123",
        revenu_annuel=150000.0,
        obligations_annuelles=50000.0,
        score_credit=780
    )

    propriete = fabrique.creer_propriete(
        adresse=Adresse(
            numero_porte="1504",
            numero_rue="1800",
            nom_rue="Bleury",
            ville="Calgary",
            province=ProvinceOuTerritoire.ALBERTA,
            code_postal="J3H3G5"
        ),
        valeur_marche=400000.0
    )

    return fabrique.creer_demande_pret(
        propriete=propriete,
        demandeur_pret=demandeur,
        prix_achat=405000.0,
        montant_mise_de_fonds=5000.0
    )

def demande_pret_non_non_non():
    demandeur = fabrique.creer_demandeur_pret(
        prenom="Donald",
        nom="Trump",
        nas="666-666-666",
        revenu_annuel=600000.0,
        obligations_annuelles=83000000.0,
        score_credit=520
    )

    propriete = fabrique.creer_propriete(
        adresse=Adresse(
            numero_porte="",
            numero_rue="325",
            nom_rue="Bay Street",
            ville="Toronto",
            province=ProvinceOuTerritoire.ONTARIO,
            code_postal="M5H 4G3"
        ),
        valeur_marche=100000.0
    )

    return fabrique.creer_demande_pret(
        propriete=propriete,
        demandeur_pret=demandeur,
        prix_achat=45000000.0,
        montant_mise_de_fonds=5000.0
    )

def demande_pret_oui_non_oui():
    demandeur = fabrique.creer_demandeur_pret(
        prenom="François",
        nom="Legault",
        nas="555-555-555",
        revenu_annuel=275000.0,
        obligations_annuelles=75000.0,
        score_credit=765
    )

    propriete = fabrique.creer_propriete(
        adresse=Adresse(
            numero_porte="",
            numero_rue="20",
            nom_rue="Alströmergatan",
            ville="Stockholm",
            province=ProvinceOuTerritoire.NEWFOUNDLAND_AND_LABRADOR,
            code_postal="M5H 4G3"
        ),
        valeur_marche=480000000.0
    )

    return fabrique.creer_demande_pret(
        propriete=propriete,
        demandeur_pret=demandeur,
        prix_achat=450000000.0,
        montant_mise_de_fonds=275000000.0
    )

    